src/surface.rs

```diff
@@ -0,0 +1,71 @@
+use na::Vector3 as Vec3;
+
+use std::collections::HashSet;
+
+#[derive(Hash, PartialEq, Eq)]
+pub struct Surface {
+    pub pixels: Vec<u8>,
+    pub w: u32,
+    pub h: u32
+}
+
+#[derive(Hash, PartialEq, Eq)]
+pub struct Division<'a> {
+    pub x0: u32,
+    pub y0: u32,
+    pub w: u32,
+    pub h: u32,
+    pub surf: &'a Surface,
+}
+
+impl Surface {
+    pub fn new(w: u32, h: u32) -> Surface {
+        let depth = 3;
+        let cap = (h * w * depth) as usize;
+        let mut p = Vec::with_capacity(cap);
+        unsafe {
+            p.set_len(cap);
+        }
+        let surf = Surface { w: w, h: h, pixels: p };
```

Comment:
The field initialization can be replaced with just `Surface { w, h, pixels: p }` if the names are the same.

Reply:
Cool, didn't know that.